### ISSUE TITLE
leading/trailing comments are messed up

### ISSUE THREAD

**Issue by DmitrySoshnikov** [babel/babel#2072]
Since recent update (5.8.5), we noticed `leadingComments` and `trailingComments` are in incorrect positions, which is also noticeable in printer:

```js
/**
 * @param {string} x
 */
function foo(/*stirng*/x) /*number*/ {
  return +x;
}
```

Generates:

```js
/**
 * @param {string} x
 */
"use strict";

function foo(x) /*number*/{
  return +x;
}
/*stirng*/
```

**Comment by sebmck** [comment]
Interestingly the comment is properly attached in the AST so the cause of this isn't very obvious. Will need to do some further investigation.

```
> require("babylon").parse("function foo(/*string*/x) {}").program.body[0].params[0].leadingComments[0];
{ type: 'CommentBlock',
  value: 'string',
  start: 13,
  end: 23,
  range: [ 13, 23 ] }
```

**Comment by DmitrySoshnikov** [comment]
Hm, I'll double-check, but seems I saw leading/trailing comments incorrectly appeared on AST, since my AST comparison tests failed.

**Comment by sebmck** [comment]
I fixed this, although I'm not sure how.

**Comment by sebmck** [comment]
Oh cool, it fixed #2068 too. I'll add a metric ton of comment attachment tests and steal some from some other parsers/projects too.

**Comment by DmitrySoshnikov** [comment]
Thanks! Mind publishing a new version please?

**Comment by sebmck** [comment]
I'll try. Current laptop is **really** slow which causes some tests to time out which means I can't push out any new releases.

**Comment by sebmck** [comment]
Whoop whoop, third time's the charm. Published 5.8.6.

**Comment by sebmck** [comment]
Wait, the change I made seems to have been reverted when a merge conflict. Will need to fix again.

**Comment by sebmck** [comment]
Fixed as of 5.8.8. Sorry for the spam.

**Comment by DmitrySoshnikov** [comment]
No worries, thanks a lot!
